flocker/node/_loop.py

```diff
@@ -300,10 +307,54 @@ def __init__(self, reactor, deployer):
         self.deployer = deployer
         self.cluster_state = None
 
+        # Save last known local state, and AMP client we sent it to.
+        # Since only one control service can exist at the moment, there can
+        # no more than one AMP client at any given point in time.
+        self.lock = threading.Lock()
+        self.last_sent_local_state = None
+        self.last_sent_client = None
+
     def output_STORE_INFO(self, context):
         self.client, self.configuration, self.cluster_state = (
             context.client, context.configuration, context.state)
 
+    def _maybe_send_state_to_control_service(self, state_changes):
+        """
+        If current state_changes differ from last sent local state,
+        transmit the new state to control agent.
+
+        :param state_changes: State to send to the control service.
+        :type state_changes: tuple of IClusterStateChange
+        """
+        if (self.last_sent_local_state != state_changes or
+           self.last_sent_client != self.client):
```

Comment:
> I wonder if it would make more sense to just clear the last sent state when we get a new client, rather than trying to track the client?

Yes, that seems cleaner. Testing out the change. Thanks!